index.js

```diff
@@ -14,30 +13,74 @@ const dbName = 'hot-dog'
 const collectionName = 'dogs'
 
 app.get('/dogs', (req, res) => {
-    mongoClient.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true }, (err, client) => {
-        if (err) {
-            return res.status(500).json({
-                success: false,
-                message: 'Error connecting to database.',
-                data: []
-            })
-        }
-        let db = client.db(dbName)
-        let collection = db.collection(collectionName)
-        collection.find({}).toArray((err, docs) => {
-            if (err) {
-                return res.status(500).json({
-                    success: false,
-                    message: 'Error getting data from database.',
-                    data: []
+    mongoClient.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
+        .then( client => {
+            let db = client.db(dbName)
+            let collection = db.collection(collectionName)
+            collection.find({}).toArray()
+                .then((docs) => {
+                    return composeResponseJson(res, 200, true, 'Data retrieved successfully', docs)
+                })
+                .catch (err => {
+                    console.log(err)
+                    return composeResponseJson(res, 500, false, 'Error getting data from database.', [])
+                })
+                .finally( () => {
+                    client.close()
                 })
-            }
-            res.status(200).json({
-                success: true,
-                message: 'Data has been successfully retrieved and sent.',
-                data: docs
+        }).catch( err => {
+            console.log(err)
+            return composeResponseJson(res, 500, false, 'Server error.', [])
+        })
+})
+
+app.post('/dogs/:id/wins', (req, res) => {
+    const id = req.params.id
+    const regex = RegExp('[0-9a-f]{24}')
+    if (regex.exec(id) === null) {
+        return composeResponseJson(res, 400, false, 'Invalid winner ID.', [])
+    }
+    mongoClient.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
+        .then( client => {
+            const db = client.db(dbName)
+            const collection = db.collection(collectionName)
+            declareChampion(collection, id, function (result) {
+                client.close()
+                if (result.success === true) {
+                    return composeResponseJson(res, 200, true, 'Dog victory recorded', [result.result.result])
+                } else {
+                    return composeResponseJson(res, 400, false, 'Not able to update record.', [])
+                }
             })
-            client.close()
+        }).catch( err => {
+            console.log(err)
+            return composeResponseJson(res, 500, false, 'Server error.', [])
         })
-    })
 })
+
+const declareChampion = function (collection, id, callback) {
```

Comment:
this and the function below are declared in different ways, please can we keep it consistent

Reply:
done